### PR TITLE
Auto create user on sign in fail + email submission MVP

### PR DESCRIPTION
**Pull request by dcgoss**
This PR solves two issues:
1. If using the stored cookie to sign in fails, the site automatically
creates a new user rather than just failing. This is much more graceful.
2. When a classifier is submitted, a prompt to enter an email where the
classifier to be sent to is shown to the user.

When the user clicks Submit, here’s what happens:
1. Check for whether the User has an email (for now, I have it set so
that you always enter an email because there is currently no way to
update an email on a user elsewhere if you make a typo or something)
2. Prompt the user to enter an email
3. Validate email, show error banner if validation fails
4. If validation passes, make a PUT request to update the User’s email
and then proceed with submitting the classifier

The implementation may be a little clunky as this is my first time using Angluar and just wanted to get it done.

### CODE REVIEW

**Comment by dcgoss** [comment]
@rzelayafavila I can't add you as an official reviewer and I'm not even sure this tag will notify you but here goes

**Comment by dcgoss** [comment]
Resolves https://github.com/cognoma/frontend/issues/97

File: app/js/components/common/progressIndicatorBar/progressIndicatorBar.component.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ const progressIndicatorBarComponent = {
                 if(goToStep == undefined || goToStep.action != undefined) return;
 
 
-                // get the index of the selcted step
+                // get the index of the selected step
```
Comment (rzelayafavila) [inline app/js/components/common/progressIndicatorBar/progressIndicatorBar.component.js]: Good catch! :+1:

File: app/js/dataModels/DiseaseModel.js [hunk 2]
```diff
@@ -102,8 +102,7 @@ function factoryWrapper($log, _, $q, $timeout, $http, AppSettings){
   // simple math for the number of samples without a mutated gene 
   // matching the user selected genes
   DiseaseModel.prototype._setNegatives = function(){
-    
-     this.negatives = this.positives ?  (this.samples - this.positives) : null;
+     this.negatives = this.samples - this.positives;
```
Comment (rzelayafavila) [inline app/js/dataModels/DiseaseModel.js]: Hmmm interesting, so we know that `this.positives` will always be defined?

File: app/js/factories/UserAuth.js [hunk 3]
```diff
@@ -25,14 +25,22 @@ function UserAuth(UserResourceService, $cookies,$log) {
     	return new Promise((resolve)=>{
           
         if(userSlug || this._getUserFromStorage()){
-
           this._authenticateUser(userSlug)
-                .then(authedCookieUser=>{
-                  resolve(authedCookieUser);
+            .catch(error => {
+              $log.log(`login:: login failed with slug ${userSlug}, creating new user`);
```
Comment (rzelayafavila) [inline app/js/factories/UserAuth.js]: Oh interesting, I'm not familiar with this `::` syntax - does it do anything in particular or just print them out?

File: app/js/dataModels/DiseaseModel.js (hunk 2 above)
Comment (dcgoss) [inline app/js/dataModels/DiseaseModel.js]: It seemed to me from looking at this: https://github.com/dcgoss/frontend/blob/e722e4fbddcd7f53fef29e9d6e797525ab1e6b45/app/js/dataModels/DiseaseModel.js#L16-L20 that everything is set to 0 to begin with

File: app/js/factories/UserAuth.js (hunk 3 above)
Comment (dcgoss) [inline app/js/factories/UserAuth.js]: I think it just prints it out, kind of gives it like a mini-title

File: app/js/components/common/progressIndicatorBar/progressIndicatorBar.component.js (hunk 1 above)
Comment (dcgoss) [inline app/js/components/common/progressIndicatorBar/progressIndicatorBar.component.js]: https://www.jetbrains.com/pycharm/ can take all the credit

File: app/js/factories/UserAuth.js (hunk 3 above)
Comment (rzelayafavila) [inline app/js/factories/UserAuth.js]: Ah, gotcha

File: app/js/dataModels/DiseaseModel.js (hunk 2 above)
Comment (rzelayafavila) [inline app/js/dataModels/DiseaseModel.js]: Gotcha, ok, makes sense.

**Pull request merged**